### PR TITLE
Trade Component Refactor

### PR DESCRIPTION
**Pull request by nalancer08**
### Type of change

- [ ] Chore
- [X] Feature
- [X] Bug fix

### Linked issue

- https://cybrid.atlassian.net/browse/CYB-1033
- https://cybrid.atlassian.net/browse/CYB-1036 (In Trade Component)

### Why

Refactor component and added this new features:

- Last UI design revision
- Integration of Accounts balance inside the component
- New way of testing
- Remove TaskSchedulers from get Quote and use new class of Polling
- Correct use of SOLID inside the component
- Correct way to join ListPrices Component and TradeComponent
- Fix Image Downloader to avoid possible issues with USDC and new trading assets

### Missed

This is the part 1 of the refactor, is missing this:

- UI Popup Confirmation
- Unit Testing of the ViewModel's
- Fix version of release
- Add new tests in E2E

### Evidence

https://user-images.githubusercontent.com/7268597/218788599-193df894-4911-4540-865c-5d2ebed6d71b.mp4

### CODE REVIEW

File: CybridSDK/Components/Trade/_TradeViewController.swift [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+                             toItem: nil,
+                             attribute: .notAnAttribute,
+                             constant: Constants.Button.height)
+  }
+}
+
+// MARK: - Data Bindings
+
+extension _TradeViewController {
+  /*func bindViewModel() {
```
Comment (brent-cybrid) [inline CybridSDK/Components/Trade/_TradeViewController.swift:238]: Remove

File: CybridSDK/Framework Extensions/UIView/UIView+Constraints.swift [hunk 2]
```diff
@@ -226,4 +226,93 @@ public extension UIView {
...
+                            toItem: nil,
+                            attribute: .notAnAttribute,
+                            constant: width)
+            self.constraint(attribute: .height,
+                            relatedBy: .equal,
+                            toItem: nil,
+                            attribute: .notAnAttribute,
+                            constant: height)
+
+            // -- Third
```
Comment (brent-cybrid) [inline CybridSDK/Framework Extensions/UIView/UIView+Constraints.swift:271]: Third before Second?

File: CybridSDKTests/ViewModel Tests/TradeViewModelTests.swift [hunk 3]
```diff
@@ -159,9 +159,9 @@ class TradeViewModelTests: XCTestCase {
     XCTAssertEqual(viewModel.numberOfComponents(in: pickerView), 1)
     XCTAssertEqual(viewModel.pickerView(pickerView, numberOfRowsInComponent: 0), 2)
     XCTAssertEqual(viewModel.pickerView(pickerView, titleForRow: 0, forComponent: 0), "Bitcoin")
-  }
+  }*/
```
Comment (brent-cybrid) [inline CybridSDKTests/ViewModel Tests/TradeViewModelTests.swift:162]: Remove

File: CybridSDKTests/ViewModel Tests/TradeViewModelTests.swift [hunk 4]
```diff
@@ -174,7 +174,7 @@ class TradeViewModelTests: XCTestCase {
 
     // Then
     XCTAssertEqual(viewModel.cryptoCurrency.value?.asset.name, "Ethereum")
-  }
+  }*/
```
Comment (brent-cybrid) [inline CybridSDKTests/ViewModel Tests/TradeViewModelTests.swift:177]: Remove

File: codecov.yml [hunk 5]
```diff
@@ -20,8 +20,8 @@ ignore:
   - "**/*Modal.swift"
   - "**/*ViewController.swift"
   - "**/*ViewController+Extensions.swift"
-  - "CybridSDK/Components/Trade/TradeViewModel.swift"
-  - "CybridSDK/Components/Crypto Price List/CryptoPriceModel.swift"
+  - "CybridSDK/Components/Trade/View/TradeViewModel.swift"
+  - "CybridSDK/Components/ListPrices/View/ListPricesViewModel.swift"
```
Comment (brent-cybrid) [inline codecov.yml:24]: Why aren't we testing the model?

**Review by brent-cybrid — COMMENTED** [review]
Looks like there are codecov issues to resolve.

File: codecov.yml (hunk 5 above)
Comment (nalancer08) [reply codecov.yml:24]: Is under creation! This is part 1 of 2 !! The second part gonna be test!

File: CybridSDK/Components/Trade/_TradeViewController.swift (hunk 1 above)
Comment (nalancer08) [reply CybridSDK/Components/Trade/_TradeViewController.swift:238]: Done!

File: CybridSDKTests/ViewModel Tests/TradeViewModelTests.swift (hunk 3 above)
Comment (nalancer08) [reply CybridSDKTests/ViewModel Tests/TradeViewModelTests.swift:162]: Done!

File: CybridSDKTests/ViewModel Tests/TradeViewModelTests.swift (hunk 4 above)
Comment (nalancer08) [reply CybridSDKTests/ViewModel Tests/TradeViewModelTests.swift:177]: Done!

**Review by brent-cybrid — APPROVED** [review]

**Pull request merged**